### ISSUE TITLE
ini path issue for plugins

### ISSUE THREAD

**Issue by Firehawke** [mamedev/mame#6315]
I'll try to shorten this as much as possible. I'm currently on a Linux older laptop for a bit, and I ran into an interesting problem with path handling in the INI file.
Most of the settings will respect ~, but plugins seems to choke when it's used.

With 
`luginspath               /home/firehawke/Projects/mame/plugins`
everything works as it should.

With
`pluginspath               ~/Projects/mame/plugins`
you get the following errors:
```
[LUA ERROR] in run: /home/firehawke/Projects/mame/plugins/boot.lua:41: module 'data' not found:
        no field package.preload['data']
        no file '~/Projects/mame/plugins/data.lua'
        no file '~/Projects/mame/plugins/data/init.lua'
        no file '/usr/local/lib/lua/5.3/data.so'
        no file '/usr/local/lib/lua/5.3/loadall.so'
        no file './data.so
```
It seems that individual files parse correctly, but scanning the subdirectory fails in this situation.
I wouldn't call it a CRITICAL bug since there's an easy workaround for an individual user, but it is a surprising pitfall and should at least be documented until the day we can fix it. Thus this issue.

**Comment by ajrhacker** [comment]
This appears to be an actual bug related to a bit of architectural chaos. Normally, `~` is expanded only by shells, so MAME needs to do a little extra work in `osd_subst_env` to expand tildes (as well as environment variables) in path definitions. The source of the trouble here appears to be the fact that while `osd_subst_env` is used when loading `boot.lua` and other scripts, `boot.lua` itself uses its own Lua function instead which does not perform any tilde expansion.

**Comment by ajrhacker** [comment]
I've added `emu.subst_env`, which seems to succeed as a drop-in substitute for `lfs.env_replace` in `boot.lua`. I'll leave the actual substitution to someone else.

**Comment by cuavas** [comment]
This is addressed by 13910382a5e846677f30d1645bfb44e4983de557 (#9859).  Environment substitution is handled in the options system itself now.
